docs/normalize-intervals.rst

```diff
@@ -0,0 +1,827 @@
+function*. A `table function`_ is a function that produces a set of rows.
+
+For example, this query generates null values for every minute in the past 24
+hours:
+
+.. code-block:: python
+
+    def null_by_minute_24h():
+        sql = '''
+            SELECT
+                "time",
+                NULL as "distance"
+            FROM
+                generate_series(
+                    date_trunc('minute', CURRENT_TIMESTAMP) - INTERVAL '24 hours',
+                    date_trunc('minute', CURRENT_TIMESTAMP), '1 minute'::INTERVAL
+                ) AS series("time")
+         '''
+        data = pandas.read_sql(sql, 'crate://localhost:4200', parse_dates={'time': 'ms'})
+        return data
+
+Test the function, like so:
+
+.. code-block:: python
+
+    null_by_minute_24h()
+
+.. csv-table::
+    :header: "", "time", "distance"
+    :widths: auto
+
+    "0", "2020-06-10 11:09:00", "None"
+    "1", "2020-06-10 11:10:00", "None"
+    "2", "2020-06-10 11:11:00", "None"
+    "3", "2020-06-10 11:12:00", "None"
+    "4", "2020-06-10 11:13:00", "None"
+    "…", "…", "…"
+    "1436", "2020-06-11 11:05:00", "None"
+    "1437", "2020-06-11 11:06:00", "None"
+    "1438", "2020-06-11 11:07:00", "None"
+    "1439", "2020-06-11 11:08:00", "None"
+    "1440", "2020-06-11 11:09:00", "None"
+
+Plot the data:
+
+.. code-block:: python
+
+    plot(null_by_minute_24h())
+
+.. image:: _assets/img/normalize-intervals/null-by-minute-24h.png
+
+This plot displays null values for a full 24 hour period.
+
+Conceptually, all that remains is to combine this null plot with the plot that
+includes your resampled data.
+
+
+.. _ni-bring-together:
+
+Bring it all together
+~~~~~~~~~~~~~~~~~~~~~
+
+To combine the null data with your resampled data, you can write a new query
+that performs a left `inner join`_, as per the previous :ref:`introductions
+<ni-interpolate>`:
+
+.. code-block:: python
+
+    def data_24h():
+        # From <https://www.latlong.net/>
+        berlin_position = [52.520008, 13.404954]
+        # Returns distance in kilometers (division by 1000)
+        sql = '''
+            SELECT
+                "time",
+                count(*) - 1 AS "records",
+                AVG(distance("iss"."position", {}) / 1000) AS "distance"
+            FROM
+                generate_series(
+                    date_trunc('minute', CURRENT_TIMESTAMP) - INTERVAL '24 hours',
+                    date_trunc('minute', CURRENT_TIMESTAMP), '1 minute'::INTERVAL
+                ) AS series("time")
+            LEFT JOIN
+                "doc"."iss" AS "iss"
+            ON
+                date_trunc('minute', "iss"."timestamp") = "time"
+            GROUP BY
+                "time"
+            ORDER BY
+                "time" ASC
+        '''.format(berlin_position)
+        data = pandas.read_sql(sql, 'crate://localhost:4200', parse_dates={'time': 'ms'})
+        return data
+
+Here's what's going on above:
```

Comment:
Somehow I felt that the phrasing here doesn't match with the overall style

Reply:
yeah I struggled with this lol!! /o\

do you have a suggestion?

Reply:
"In the code above:"

?